docs/proposals/clusters.md

```diff
@@ -0,0 +1,214 @@
+
+
+## Work Scope
+
+This update will require changes to the Data Model definitions, parsing process, and state engine logic. All of these currently assume all inventory will be collected in one homogenous group.
+
+## Estimated Development Time (SWAG)
+
+Medium complexity. Approx. one month with two developers for implementation, testing, and delivery.
+
+
+
+# Separated Configuration/Inventory
+
+After hosts have been separated into clusters, we consider the process for separating the inventory from the defined configurations. The goal is to contain all inventory, cluster, and site information in one location, and maintain workflow and associated configurations in a separate location.
+
+## Requirements
+
+- Provide per-cluster info for linking to remote configuration
+  - Github configuration for pulling from a remote repository
+  - Local directory configuration for referencing on-disk configs
+- Clusters contain metadata that augments site metadata
+
+## Design
+
+The current logical layout for a data model is:
+
+```
+├── assets
+│   └── Config files, templates
+├── boot
+│   └── Pair configs + OSes, and metadata
+├── os
+│   └── OS kernel, initrd, cmdline information
+├── sites
+│   └── site-name
+│       ├── env.yml <- Specifies agent configurations, site-wide metadata, subnet information
+│       └── inventory.yml <- Host definitions deliminated as individual YAML documents 
+└── workflows
+    └── Chain multiple boots together, inter-workflow dependencies and rollout policy
+```
+
+In the current layout, all site inventory and metadata is stored under `sites`. Everything else is configuration.
+
+
+Separating the data model will place all site inventory and metadata into one location, and all configuration into another:
+
+`Location 1: Inventory`
+```
+└── sites
+    ├── site-one
+    │   ├── env.yml <- Specifies agent configurations, site-wide metadata, subnet information
+    │   ├── cluster-one.yml <- Host definitions deliminated as individual YAML documents 
+    │   └── cluster-two.yml <- Host definitions deliminated as individual YAML documents 
+    └── site-two
+        ├── env.yml <- Specifies agent configurations, site-wide metadata, subnet information
+        └── inventory.yml <- Host definitions deliminated as individual YAML documents 
+```
+
+`Location 2: Configuration`
+```
+├── assets
+│   └── Config files, templates
+├── boot
+│   └── Pair configs + OSes, and metadata
+├── os
+│   └── OS kernel, initrd, cmdline information
+└── workflows
+    └── Chain multiple boots together, inter-workflow dependencies and rollout policy
+```
+
+Additionally, each inventory file will specify a configuration document with the details required for fetching the desired configuration from a remote location. For example: 
+
+`site-one/cluster-one.yml`
+```
+---
+# cluster config document
+config:
```

Comment:
IMO it'd make sense to have some sort of typing indicator on this document indicating that it contains cluster configuration and connecting it to the cluster definition, rather than having it be implicitly based on being the first document.